File: Javascript/Arrays & Hashing/Two-Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const numsMap = new Map();

  for (currentIndex in nums) {
    const currentVal = nums[currentIndex];
    const difference = target - currentVal;
    if (numsMap.has(difference)) {
      return [currentIndex, numsMap.get(difference)];
    }

    numsMap.set(currentVal, currentIndex);
  }
};

File: Python/Arrays & Hashing/Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsMap = {}

        for currentIndex, currentVal in enumerate(nums):
            difference = target - currentVal
            if difference in numsMap:
                return [currentIndex, numsMap[difference]]

            numsMap[currentVal] = currentIndex
